bot/exts/utilities/bookmark.py

```diff
@@ -23,6 +23,40 @@
 )
 
 
+class BookmarkTitleSelectionForm(discord.ui.Modal):
+    """
+    The form where a user can fill in a custom title for their bookmark & submit it.
+
+    This form is only available when the command is invoked from a context menu.
+    """
+
+    bookmark_title = discord.ui.TextInput(
+        label="Choose a title for you bookmark (optional)",
+        placeholder="Type your bookmark title here",
+        default="Bookmark",
+        max_length=50,
+        min_length=0,
+        required=False
+    )
+
+    def __init__(
+            self,
+            message: discord.Message,
+            action_bookmark_function: Callable[[discord.TextChannel, discord.Member, discord.Message, str], None],
+    ):
+        super().__init__(timeout=1000, title="Name your bookmark")
+        self.message = message
+        self.action_bookmark = action_bookmark_function
+
+    async def on_submit(self, interaction: discord.Interaction) -> None:
+        """Sends the bookmark embed to the user with the newly chosen title."""
+        title = self.bookmark_title.value or self.bookmark_title.default
+        await self.action_bookmark(interaction.channel, interaction.user, self.message, title)
+        view = SendBookmark(self.action_bookmark, interaction.user, interaction.channel, self.message, title)
```

Comment:
Since this message is ephemeral, I don't think there's a need for the button that says "Click this to get your very own bookmark" or whatever it was. The whole point of the button was that multiple people wouldn't be running the .bookmark command in succession to the same message and flooding the channel - but with ephemeral messages, that's not an issue. So I think this button would just lead to more confusion and is unnecessary.

Reply:
This one has also been resolved in cf4903561a222de9b7413c406b5d6e186f977332

I just happened to make a new static method that builds an embed that's only used when the context menu command is invoked